packages/api/src/promise/Api.ts

```diff
@@ -249,4 +249,73 @@ export default class ApiPromise extends ApiBase<'promise'> {
       }) as UnsubscribePromise;
     } as StorageEntryPromiseOverloads;
   }
+
+  protected exposeRpc (rpc: DecoratedRpc<'rxjs'>): DecoratedRpc<'promise'> {
+    return Object.keys(rpc).reduce((result, _sectionName) => {
+      const sectionName = _sectionName as keyof DecoratedRpc<'rxjs'>;
+
+      result[sectionName] = Object.keys(rpc[sectionName]).reduce((section, methodName) => {
+        const method = rpc[sectionName][methodName];
+        section[methodName] = this.exposeMethod(method);
+
+        return section;
+      }, {} as DecoratedRpc$Section<'promise'>);
+
+      return result;
+    }, {} as DecoratedRpc<'promise'>);
+  }
+
+  protected exposeMulti (multi: QueryableStorageMulti<'rxjs'>): QueryableStorageMulti<'promise'> {
+    return this.exposeMethod(multi) as QueryableStorageMulti<'promise'>;
+  }
+
+  protected exposeExtrinsics (_extrinsics: SubmittableExtrinsics<'rxjs'>): SubmittableExtrinsics<'promise'> {
+    const creator = this.exposeMethod(_extrinsics);
+
+    return Object.keys(_extrinsics).reduce((result, sectionName) => {
+      const section = _extrinsics[sectionName];
+
+      result[sectionName] = Object.keys(section).reduce((result, methodName) => {
+        result[methodName] = this.exposeMethod(section[methodName]) as any;
+        result[methodName].meta = section[methodName].meta;
+        result[methodName].method = section[methodName].method;
+        result[methodName].section = section[methodName].section;
+        result[methodName].toJSON = section[methodName].toJSON;
+        if ('callIndex' in section[methodName]) {
+          result[methodName].callIndex = section[methodName].callIndex;
+        }
+
+        return result;
+      }, {} as SubmittableModuleExtrinsics<'promise'>);
+
+      return result;
+    }, creator as any/*?*/ as SubmittableExtrinsics<'promise'>);
+  }
+
+  protected exposeStorage (storage: QueryableStorage<'rxjs'>): QueryableStorage<'promise'> {
+    return Object.keys(storage).reduce((result, sectionName) => {
+      const section = storage[sectionName];
+
+      result[sectionName] = Object.keys(section).reduce((result, methodName) => {
+        result[methodName] = this.exposeMethod(section[methodName]) as any;
+        result[methodName].at = this.exposeMethod(section[methodName].at);
+        result[methodName].hash = this.exposeMethod(section[methodName].hash);
+        result[methodName].key = section[methodName].key;
+        result[methodName].multi = this.exposeMethod(section[methodName].multi);
+        result[methodName].size = this.exposeMethod(section[methodName].size);
+        result[methodName].creator = section[methodName].creator;
+        (result[methodName] as any).meta = (section[methodName] as any).meta;
```

Comment:
We `as any` a lot in here and above, it is a danger sign that we don't actually know what we have and are working with.